### PR TITLE
Merge upstream changes

### PR DESCRIPTION
**Pull request by ClearlyClaire**
## Conflicts

None

## Ported changes

- b46125224c26bc0144659b4319ea416b01289961
- d1b057a0ac41a5756b6a416d2f290b086cb59220, 6e8711ff91e01212ac7c82de6349fa2b143582e2 (from #2150)

## Skipped changes

- 16da4a09e8bd7ce383487530b97ffa7dfa59b960 (issue does not affect glitch flavor)

### CODE REVIEW

File: app/javascript/flavours/glitch/features/compose/components/options.jsx [hunk 1]
```diff
@@ -83,8 +83,11 @@ const messages = defineMessages({
   },
 });
 
-@connect((state, { name }) => ({ checked: state.getIn(['compose', 'advanced_options', name]) }))
-class ToggleOption extends ImmutablePureComponent {
+const mapStateToProps = (state, { name }) => ({
+  checked: state.getIn(['compose', 'advanced_options', name]),
+});
+
+class ToggleOptionIntl extends ImmutablePureComponent {
```
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/features/compose/components/options.jsx:90]: For some reason this has reverted back to `ToggleOptionIntl`

File: app/javascript/flavours/glitch/features/compose/components/poll_form.jsx [hunk 2]
```diff
@@ -21,8 +21,7 @@ const messages = defineMessages({
   days: { id: 'intervals.full.days', defaultMessage: '{number, plural, one {# day} other {# days}}' },
 });
 
-@injectIntl
-class Option extends React.PureComponent {
+class OptionIntl extends React.PureComponent {
```
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/features/compose/components/poll_form.jsx:24]: This should be `OptionImpl`

File: app/javascript/flavours/glitch/features/getting_started_misc/index.jsx [hunk 3]
```diff
@@ -68,3 +66,5 @@ class gettingStartedMisc extends ImmutablePureComponent {
   }
 
 }
+
+export default connect()(injectIntl(gettingStartedMisc));
```
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/features/getting_started_misc/index.jsx:70]: Can we fix the casing of this class while at it?

File: app/javascript/flavours/glitch/features/list_adder/components/list.jsx [hunk 4]
```diff
@@ -67,3 +65,5 @@ class List extends ImmutablePureComponent {
   }
 
 }
+
+export default connect(MapStateToProps, mapDispatchToProps)(injectIntl(List));
```
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/features/list_adder/components/list.jsx:69]: Can we fix the casing of `MapStateToProps` while at it?

**Review by Plastikmensch — CHANGES_REQUESTED** [review]

File: app/javascript/flavours/glitch/features/compose/components/options.jsx (hunk 1 above)
Comment (ClearlyClaire) [reply app/javascript/flavours/glitch/features/compose/components/options.jsx:90]: No, this wasn't on purpose. Fixing it.

File: app/javascript/flavours/glitch/features/getting_started_misc/index.jsx (hunk 3 above)
Comment (ClearlyClaire) [reply app/javascript/flavours/glitch/features/getting_started_misc/index.jsx:70]: Sure.

**Review by Plastikmensch — APPROVED** [review]

**Pull request merged**